lib/srv/db/sqlserver/connect_test.go

```diff
@@ -116,3 +125,257 @@ func TestConnectorSelection(t *testing.T) {
+c2NhbGUubmV0MTQwMgYDVQQDEytjZXJlYnJvLmFsaXN0YW5pcy5naXRodWIuYmV0
+YS50YWlsc2NhbGUubmV0MTAwLgYDVQQFEyczMjA0NTY3ODIyNjAyNTg5MjI3OTU5
+NjQ3NDExMjk1NDgzMDc2ODAwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIB
+AQDJVqHTgx9pdPHCrDJ0UtbZMVL/xhihuR44AY8aqSebJbKc/WrYLIJxqO1q8L4c
+B+sfblIMMz/Em1IZ3ZF7AajiJFSn8VfGx5xtxC06YWPY3HfflcuY5kGVWtYl8ReD
+7j3FJjNq4Rvv+NoYwmQXYw6Nwu90cWHerDY3G0fQOsjgUAipnTS4+/H36pBakNoK
+9pipl3Kb6YVtjdxY6KY0gSy0k8NiRUx8sCpxJOwfUSAvtsGd1tw1388ZfWr2Bl2d
+st2H+q1ozLZ3IQXSgSl6s63JmvWpsElg8+nXZKB3CNTIhrOvvyV33Ok5uAQ44nel
+vLy5r3o2OguPjvC+SrkHn1avAgMBAAGjQjBAMA4GA1UdDwEB/wQEAwIBpjAPBgNV
+HRMBAf8EBTADAQH/MB0GA1UdDgQWBBR0fa5/2sVguUfn8MHmC7DoFl58fzANBgkq
+hkiG9w0BAQsFAAOCAQEAAOEBowwaigoFG3rxM5euIyfax2gWPXN63YF3vd5IN75C
+gzimkq9c6MRsvaS053xbRF5NncectmBzTY3WQscJ30+tHD84fA5VQCt//lA+G9gi
+g8Co+YPraQe8kbZEcAFceGpWrKjCEwiWlrlM56VfmKmGws21N/PBIb5aO0aEHuWs
+HOhXH/n0dKrb7IJcpUh0/w02qiUQ6I0usjGwRlE3xkPyWgEkKUcy+eBrfVVV++8e
+HDKyflZ05nt/zvM6W/WIeMI7VMPw/Ryr7iynMqAYAhJhTFKdSwuNLDY8eFbOUnbw
+21sZcc/b5g+C9N+0lbFxUUF99bt6jLOVUwpR7LRP2g==
+-----END CERTIFICATE-----`
+
+	krb5Conf = `[libdefaults]
+ default_realm = example.com
+ rdns = false
+
+
+[realms]
+ example.com = {
+  kdc = host.example.com
+  admin_server = host.example.com
+  pkinit_eku_checking = kpServerAuth
+  pkinit_kdc_hostname = host.example.com
+ }`
+)
+
+type mockClusterName struct {
+}
+
+func (m mockClusterName) GetKind() string {
+	return ""
+}
+
+func (m mockClusterName) GetSubKind() string {
+	return ""
+}
+
+func (m mockClusterName) SetSubKind(s string) {
+}
+
+func (m mockClusterName) GetVersion() string {
+	return ""
+}
+
+func (m mockClusterName) GetName() string {
+	return ""
+}
+
+func (m mockClusterName) SetName(s string) {
+}
+
+func (m mockClusterName) Expiry() time.Time {
+	return time.Time{}
+}
+
+func (m mockClusterName) SetExpiry(t time.Time) {
+}
+
+func (m mockClusterName) GetMetadata() types.Metadata {
+	return types.Metadata{}
+}
+
+func (m mockClusterName) GetResourceID() int64 {
+	return 0
+}
+
+func (m mockClusterName) SetResourceID(i int64) {
+}
+
+func (m mockClusterName) CheckAndSetDefaults() error {
+	return nil
+}
+
+func (m mockClusterName) SetClusterName(s string) {
+}
+
+func (m mockClusterName) GetClusterName() string {
+	return "TestCluster"
+}
+
+func (m mockClusterName) SetClusterID(s string) {
+}
+
+func (m mockClusterName) GetClusterID() string {
+	return ""
+}
+
+func (m mockClusterName) Clone() types.ClusterName {
+	return m
+}
+
+type mockAuth struct{}
+
+func (m *mockAuth) GenerateWindowsDesktopCert(ctx context.Context, request *proto.WindowsDesktopCertRequest) (*proto.WindowsDesktopCertResponse, error) {
+	return nil, nil
+}
+
+func (m *mockAuth) GetCertAuthority(ctx context.Context, id types.CertAuthID, loadKeys bool, opts ...services.MarshalOption) (types.CertAuthority, error) {
+	return nil, nil
+}
+
+func (m *mockAuth) GetClusterName(opts ...services.MarshalOption) (types.ClusterName, error) {
+	return &mockClusterName{}, nil
```

Comment:
Nit: To save having to write a mock cluster name type, I'd probably just have returned a real cluster name here:
```go
types.NewClusterName(types.ClusterNameSpecV2{
		ClusterName: "TestCluster",
	})
```

Reply:
Thanks! I didn't realize it was that simple. Saves me a lot of space.